src/elevenlabs/conversation.py

```diff
@@ -0,0 +1,282 @@
+            event = message["interruption_event"]
+            self.last_interrupt_id = int(event["event_id"])
+            self.audio_interface.interrupt()
+        elif message["type"] == "ping":
+            event = message["ping_event"]
+            ws.send(
+                json.dumps(
+                    {
+                        "type": "pong",
+                        "event_id": event["event_id"],
+                    }
+                )
+            )
+            if self.callback_latency_measurement and event["ping_ms"]:
+                self.callback_latency_measurement(int(event["ping_ms"]))
+        else:
+            pass  # Ignore all other message types.
+
+    def _get_wss_url(self):
+        base_url = self.client._client_wrapper._base_url
+        # Replace http(s) with ws(s).
+        base_ws_url = base_url.replace("http", "ws", 1)  # First occurrence only.
+        return f"{base_ws_url}/v1/convai/conversation?agent_id={self.agent_id}"
+
+    def _get_signed_url(self):
+        # TODO: Use generated SDK method once available.
+        response = self.client._client_wrapper.httpx_client.request(
+            f"v1/convai/conversation/get_signed_url?agent_id={self.agent_id}",
+            method="GET",
+        )
+        return response.json()["signed_url"]
+
+
+def get_default_audio_interface():
+    """Returns a default audio interface implementation.
+
+    Uses the default system input and output sound devices.
+    """
+    try:
+        import pyaudio
+    except ImportError:
+        raise ImportError("To use the default audio interface, you must install and pyaudio.")
+
+    class DefaultInterface(AudioInterface):
+        INPUT_FRAMES_PER_BUFFER = 4000   # 250ms @ 16kHz
+        OUTPUT_FRAMES_PER_BUFFER = 1000  # 62.5ms @ 16kHz
+
+        def start(self, input_callback: typing.Callable[[bytes], None]):
+            # Audio input is using callbacks from pyaudio which we simply pass through.
+            self.input_callback = input_callback
+
+            # Audio output is buffered so we can handle interruptions.
+            # Start a separate thread to handle writing to the output stream.
+            self.output_queue: queue.Queue[bytes] = queue.Queue()
+            self.should_stop = threading.Event()
+            self.output_thread = threading.Thread(target=self._output_thread)
+
+            self.p = pyaudio.PyAudio()
+            self.in_stream = self.p.open(
+                format=pyaudio.paInt16,
+                channels=1,
+                rate=16000,
+                input=True,
+                stream_callback=self._in_callback,
+                frames_per_buffer=self.INPUT_FRAMES_PER_BUFFER,
+                start=True,
+            )
+            self.out_stream = self.p.open(
+                format=pyaudio.paInt16,
+                channels=1,
+                rate=16000,
+                output=True,
+                frames_per_buffer=self.OUTPUT_FRAMES_PER_BUFFER,
```

Comment:
The docs aren't super clear on this (they clearly chose the golang way of comments 
<img width="445" alt="image" src="https://github.com/user-attachments/assets/918a19eb-8af0-4927-a93b-323ca45bd427">
)
but basically smaller buffer allows us to cut the agent faster - the stuff already in the audio buffer we sent to pyaudio can't be changed so it will play. 

I'm not entirely satisfied with it yet, could eg. do a smoother volume crossfade instead of immediate interruption. But this change should not impact the public API so I'll save that for later, what we have is good enough for first demos.